### PR TITLE
feat: use per-asset debt limit and min initial debt

### PR DESCRIPTION
**Pull request by samsiegart**
fixes: https://github.com/Agoric/agoric-sdk/issues/5066

### CODE REVIEW

**Review by turadg — CHANGES_REQUESTED** [review]

File: ui/src/contexts/Application.jsx [hunk 1]
```diff
@@ -183,6 +168,21 @@ function watchOffers(dispatch, INSTANCE_BOARD_ID) {
...
+    E(treasuryAPI).getGovernedParams({ collateralBrand: collateral.brand }),
+  ]);
+  dispatch(mergeGovernedParams([[collateral.brand, governedParams]]));
+
+  const notifier = await E(manager).getNotifier();
+  for await (const state of iterateNotifier(notifier)) {
+    console.log('asset state', collateral.brand, state);
+    dispatch(mergeVaultAssets([[collateral.brand, state]]));
+  }
+  console.log('notifier finished');
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply ui/src/contexts/Application.jsx]: Just decided to remove it instead

File: ui/src/store.js [hunk 2]
```diff
@@ -16,14 +16,16 @@ export const initial = {
   // Vault state
   treasury: /** @type { VaultState | null } */ (null),
   vaultCollateral: /** @type { CollateralInfo | null } */ (null),
+  vaultAssets: /** @type { Array<[Brand, VaultAssetState]> | null } */ (null),
+  governedParams: /** @type { Array<[Brand, Record<string, any>]> | null } */ (null),
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply ui/src/store.js]: Good idea, done

**Review by samsiegart — COMMENTED** [review]
Thanks for review! PTAL

File: ui/src/components/vault/VaultManagement/VaultManagement.jsx [hunk 3]
```diff
@@ -231,11 +235,23 @@ const VaultManagement = () => {
 
   useEffect(() => {
     if (debtAction === 'borrow') {
+      const newDebt = AmountMath.add(debt, debtDelta);
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply ui/src/components/vault/VaultManagement/VaultManagement.jsx:247]: Done

File: ui/src/components/vault/VaultManagement/VaultManagement.jsx [hunk 4]
```diff
@@ -76,12 +75,13 @@ const VaultManagement = () => {
...
-    asset,
   } = vaultToManage;
 
+  const asset = locked && new Map(vaultAssets).get(locked.brand);
+  const params = locked && new Map(governedParams).get(locked.brand);
   assert(
-    locked && debtSnapshot && asset,
-    `Can't manage vault with missing data: locked: ${locked}, debt: ${debtSnapshot}, asset: ${asset}`,
+    params && locked && debtSnapshot && asset,
+    `Can't manage vault with missing data: locked: ${locked}, debt: ${debtSnapshot}, asset: ${asset}, params: ${params}`,
```
[the remark this answers is not in the document]
Comment (turadg) [reply ui/src/components/vault/VaultManagement/VaultManagement.jsx:85]: my mistake. I read it as a boolean but it's the locked amount 👍

**Review by turadg — APPROVED** [review]
LGTM!

**Pull request merged**